Desktop

AnnGabr opened the pull request.

AnnGabr committed:
Created index file.

AnnGabr committed:
Created index file.

AnnGabr committed:
Added header.

AnnGabr committed:
Added banner.

AnnGabr committed:
Changed body from flex to block.

AnnGabr committed:
Added service section.

AnnGabr committed:
Added arrow-circle-right icon to service block.

AnnGabr committed:
Added people block in staff section.

AnnGabr committed:
Changed classes names of staff block to BEM.

AnnGabr committed:
Added panel layout.

AnnGabr committed:
Added attention block.

AnnGabr committed:
Added comments list and changed some classes names.

AnnGabr committed:
Changed classes names of company-life block.

AnnGabr committed:
Attention and company-life section css classes renamed.

AnnGabr committed:
Added news list to company-life section.

AnnGabr committed:
Changed height property of sections to min-height for more flexibility.

AnnGabr committed:
Put news component to <li> tag.

AnnGabr committed:
Changed more-icon position.

AnnGabr committed:
Added footer.

AnnGabr committed:
Changed layout of service section.

AnnGabr committed:
Changed company-life body to nowrap flex.

AnnGabr committed:
Changed attention section layout.

AnnGabr committed:
Fixed panel layout and added staff panel content.

AnnGabr committed:
Fixed some layout bugs.

AnnGabr committed:
Added page background.

AnnGabr committed:
Positioned header by psd.

AnnGabr committed:
Fixed px to px position in blocks before company-life block.

AnnGabr committed:
Positioned company-life, footer.

AnnGabr committed:
Positioned all.

AnnGabr committed:
Changed div tag to button where needed.

AnnGabr committed:
Fixed postion of some items. Refactored header.

AnnGabr committed:
Refactored banner.

AnnGabr committed:
Refactored service.

AnnGabr committed:
Refactored attention.

AnnGabr committed:
Refactored company-life, staff-panel.

AnnGabr committed:
Changed footer list to grid layout.

AnnGabr committed:
Refactored footer classes.

AnnGabr committed:
Changed colors forms to  hex. Removed min-height from services section container. Changed staff-panel section layout to grid.

ahryhoryeu commented on styles/main.css:
```diff
@@ -0,0 +1,89 @@
+html {
+    background-image: url('../resources/images/page-background.png');
+    box-sizing: border-box;
+    padding: 101px 0 90px 0;
```
Positioning of `<body>` could be done much simpler with one line of CSS. Removing these paddings and flexbox properties from `<html>` and adding `margin: 101px auto 90px` to `<body>` would perfectly do the trick.

ahryhoryeu replied:
And also, please make use of short-hand declaration. Defining `padding: 101px 0 90px 0;` is the same as `padding: 101px 0 90px;`.

ahryhoryeu commented on styles/main.css:
```diff
@@ -0,0 +1,89 @@
+html {
```
In general, this file is not following BEM in any form. It's a pile of styles for various elements. This should be fixed. Following the structure is paramount when building maintainable code.

ahryhoryeu commented on styles/main.css:
```diff
@@ -0,0 +1,89 @@
+html {
+    background-image: url('../resources/images/page-background.png');
+    box-sizing: border-box;
```
Do you think it's needed? I don't spot where it makes the difference.

ahryhoryeu commented on styles/main.css:
```diff
@@ -0,0 +1,89 @@
+    hyphens: manual;
+}
+
+.prh--white {
```
Naming classes with colors and other signs of CSS is considered a bad practice. The UI representation can be changed (I mean CSS rules), but in this case it would lead to changing the markup as well, which breaks the whole idea of seaparting CSS from markup.

ahryhoryeu commented on styles/panel.css:
```diff
@@ -0,0 +1,103 @@
+.panel {
+    display: flex;
```
Why are you mixing flex and grids in a single element? It makes no sense.

ahryhoryeu commented on styles/panel.css:
```diff
@@ -0,0 +1,103 @@
+    width: 107px;
+    background-color: #003d48;
+    color: #f77462;
+    z-index: 1;
```
Are you sure it's needed?

ahryhoryeu commented on styles/panel.css:
```diff
@@ -0,0 +1,103 @@
+
+.tab__btn {
+    height: 41px;
+    width: 96px;
```
Fully statically-sized elements are not really adaptive for dynamic content. So in case the button has more text, it will wrap instead of becoming bigger.

ahryhoryeu commented on styles/company-life.css:
```diff
@@ -0,0 +1,177 @@
+    text-align: center;
+}
+
+.life__header + .life__body {
```
Relying on HTML structure in CSS is a bad practice. Arguably `life__body` should have a margin in any case. Or, perhaps, the margin belongs to the header rather than the body.

ahryhoryeu commented on styles/company-life.css:
```diff
@@ -0,0 +1,177 @@
+    width: 95%;
+}
+
+.life__header, life__body{
```
Invalid selector `life__body`.

ahryhoryeu commented on styles/company-life.css:
```diff
@@ -0,0 +1,177 @@
+}
+
+.life {
+    margin: 54px 0 54px 0;
```
Again, please opt for shorthand properties: `margin: 54px 0;` seems much better.

ahryhoryeu commented on styles/company-life.css:
```diff
@@ -0,0 +1,177 @@
+    margin-top: 50px;
+}
+
+.news {
```
What is `news` block declaration doing in the same file as `life`?

ahryhoryeu commented on styles/attention.css:
```diff
@@ -0,0 +1,24 @@
+.attention-container {
+    width: 100%;
```
It was defined in many other places, for example in `.container`.

ahryhoryeu commented on styles/attention.css:
```diff
@@ -0,0 +1,24 @@
+}
+
+.attention__title, .attention__prh {
+    color: white;
```
You need to be consistent in colors. Advised schemes of declaring colors are: hex notation (6 digits, all lower-case); rgba, hsla. Avoid using shorthands like `white` or `cyan`, using rgb and hsl without transparency parameter (in this case it's better to opt for hex `#ffffff` notation).

ahryhoryeu commented on styles/banner.css:
```diff
@@ -0,0 +1,68 @@
+.banner-container {
+    background-image: -moz-linear-gradient( 90deg, rgb(255,255,255) 0%, rgb(243,243,243) 79%, rgb(231,231,231) 100%);
+    background-image: -webkit-linear-gradient( 90deg, rgb(255,255,255) 0%, rgb(243,243,243) 79%, rgb(231,231,231) 100%);
+    background-image: -ms-linear-gradient( 90deg, rgb(255,255,255) 0%, rgb(243,243,243) 79%, rgb(231,231,231) 100%);
```
I think a version without prefixes is needed here as well.

ahryhoryeu commented on styles/banner.css:
```diff
@@ -0,0 +1,68 @@
+.banner-container {
+    background-image: -moz-linear-gradient( 90deg, rgb(255,255,255) 0%, rgb(243,243,243) 79%, rgb(231,231,231) 100%);
```
As per comment for `attention.css`, don't use rgb notation for colors declaration.

ahryhoryeu commented on styles/banner.css:
```diff
@@ -0,0 +1,68 @@
+    align-items: flex-end;
+    flex-wrap: wrap;
+
+    margin: 52px 0 0 0;
```
Do you really need to explicitly say that all margins except for the top one are zeros, or would it be sufficient to say `margin-top: 52px`?

ahryhoryeu commented on styles/banner.css:
```diff
@@ -0,0 +1,68 @@
+    flex-wrap: wrap;
+
+    margin: 52px 0 0 0;
+    box-sizing: border-box;
```
Please review using of `box-sizing: border-box;` as it seems to be redundant in many places.

ahryhoryeu commented on styles/banner.css:
```diff
@@ -0,0 +1,68 @@
+}
+
+.banner__btn {
+    width: 129px;
```
Instead of hard-coding width and height for buttons, consider defining paddings for the text inside.

ahryhoryeu commented on styles/panel.css:
```diff
@@ -0,0 +1,103 @@
+    display: flex;
+    justify-content: flex-start;
+    width: 100%;
+    margin-bottom: -2px;
```
Consider other ways of solving this problem, for example using a pseudo-element (`::after`) to cover the gap.

ahryhoryeu commented on styles/banner.css:
```diff
@@ -0,0 +1,68 @@
+.banner__subtitle {
+    font-weight: 300;
+    font-size: 30px;
+    word-spacing: 2.3px;
```
I don't think we need to be that precise to use fractions of pixels.

ahryhoryeu commented on styles/company-life.css:
```diff
@@ -0,0 +1,177 @@
+    line-height: 46px;
+    height: 48px;
+
+    border-width: 1px 0 1px 0;
```
`border-width` has a shorthand syntax as well.

ahryhoryeu commented on styles/company-life.css:
```diff
@@ -0,0 +1,177 @@
+}
+
+.news__article {
+    display: flex;
```
For example, flexbox here is completely redundant. It doesn't place elements in some tricky order and placing elements in column is just a natural flow of elements in the documents.

ahryhoryeu commented on styles/footer.css:
```diff
@@ -0,0 +1,82 @@
+    width: 100%;
+    margin-bottom: 34px;
+
+    display: grid;
```
Opt for one thing here. If you use grids, then use grids. If you use flexbox, then you need to place elements using flexbox. Both options are completely possible here. But a mix looks really weird.

ahryhoryeu replied:
Well, I might have been referring to something different here. Nevermind, skip this.

ahryhoryeu commented on styles/footer.css:
```diff
@@ -0,0 +1,82 @@
+}
+
+.footer__underline {
+    display: flex;
```
Yey! This is the case when using flexbox is completely justified. It would be great to use appropriate tools to solve appropriate problems.

ahryhoryeu reviewed (changes requested):
Please review the comments. I'd like to point out three main guidelines:
1. Use CSScomb. It was mentioned in the task provided. Visit its site and explore what it is. Then ping me and I'll send you the configuration file.
2. There are many redundant styles which make your CSS look messy. Take some time to clean that out.
3. Flexbox everywhere is not that good. It's a specific tool designed to solve specific problems (the same applies to CSS Grids by the way). Take some time to investigate and read, for example: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Typical_Use_Cases_of_Flexbox.

AnnGabr committed:
Replaced main.css. Changed header file structure.

AnnGabr committed:
Refactored header css classes: removed redundant properties.

AnnGabr committed:
Removed flex from banner container. Replaced btn sizes to paddings.

AnnGabr committed:
Refactored service:
- changed file structure;
- removed redundant containers.

AnnGabr committed:
Refactored staff-panel:
- changed file structure;
- removed flex-boxes;
- removed fixed sizes.

AnnGabr committed:
Added paddings to staff-panel block buttons.

AnnGabr committed:
Chanded paddings of panel sub-info block.

AnnGabr committed:
Removed flex-box from attention block.

AnnGabr committed:
Refactored footer:
- removed flex-box;
- changed file structure.

AnnGabr committed:
Changed file structure of  company-life block.

AnnGabr committed:
Removed flexes from company-life block.

AnnGabr committed:
Refactored company-life block.

AnnGabr committed:
Positioned news-item and says-item content.

AnnGabr committed:
Moved styles of body and html to classes.

AnnGabr committed:
Added links.

AnnGabr committed:
Linked all styles to page.

AnnGabr committed:
Removed main with style imports.

AnnGabr committed:
Added after element to btn to hide separating line. Applied csscomb tool.

ahryhoryeu commented on styles/blocks/header/navigation.css:
```diff
@@ -0,0 +1,19 @@
+    border-left: solid 1px #e7e7e7;
+}
+
+.nav-item--first {
```
Semantically, it's more like `.nav-item--active`.

AnnGabr replied:
What about first-child?

ahryhoryeu replied:
What about it? I’m just saying that based on the UI and common sense, it’s not the first item that gets highlighted, but rather the active or current menu item.

ahryhoryeu commented on styles/blocks/header/navigation.css:
```diff
@@ -0,0 +1,19 @@
+.nav-item {
+    display: inline-block;
+
+    padding: 0 27.5px 0 25.6px;
```
Are you sure it's needed to be this perfect? Fractions of pixels seem kind of negligible to me.
Furthermore, borders of menu items do not fall in the same places as on designs.

ahryhoryeu commented on styles/blocks/banner/__content/content.css:
```diff
@@ -0,0 +1,15 @@
+.banner__content {
+    display: flex;
+
+    margin: auto;
```
This is redundant.

AnnGabr replied:
I want to change order of inner items in mobile, so I added flex here. Yes, may be I should move it to media query.

ahryhoryeu replied:
I'm talking about margin: auto here :)

ahryhoryeu commented on styles/blocks/banner/__content/content.css:
```diff
@@ -0,0 +1,15 @@
+    justify-content: center;
+}
+
+.banner__screen-img + .banner__info-container {
```
So, using this selector for positioning elements of different classes is wrong. It can be justified for using with the same classes, though.

ahryhoryeu commented on styles/blocks/banner/__content/info.css:
```diff
@@ -0,0 +1,45 @@
+    padding-bottom: 77px;
+}
+
+.banner__article {
```
Do you think that `.banner__article` deserves being a block? It can be positioned using the element class.

AnnGabr replied:
Do you mean removing `.banner__article` at all?

ahryhoryeu replied:
I mean that what is `banner__article` can be a block. But blocks shouldn’t position themselves. So I mean that there can be an _element_ `banner__article` which will only do positioning. But the article itself can be a block on its own.

ahryhoryeu commented on styles/blocks/service/service-item.css:
```diff
@@ -0,0 +1,63 @@
+.service-item__flag-img {
+    width: 50%;
+    height: 71%;
+    margin: 16% 0 0 30%;
```
Don't you think that _centering_ using margins is weird? I would advise to opt for other ways, even flex will do here.

ahryhoryeu replied:
What's better us that you wont have to use different margins for different images.

AnnGabr replied:
Say goodbye to pixel perfect=)

AnnGabr replied:
But I think size is nessesary here. So I am going to leave classes separated fo each icon, am I?

ahryhoryeu replied:
Probably. In the ideal world you’d want to have those images of the same size. If you want, you can try resizing, otherwise stick to setting the sizes.

ahryhoryeu commented on styles/blocks/service/service-item.css:
```diff
@@ -0,0 +1,63 @@
+
+.service-item__title {
+    margin-bottom: 21px;
+    margin-left: -2.1px;
```
Negative margins is a very, very bad thing. Try avoiding it at all costs. In this particular case, it's ok without this margin at all.

ahryhoryeu commented on styles/blocks/staff-panel/panel.css:
```diff
@@ -0,0 +1,99 @@
+    margin-top: 2px;
+}
+
+.tab__btn {
```
Naming doesn't correspond to the file name. Did you mean something other?

AnnGabr committed:
Removed selectors relied on html structure.

ahryhoryeu commented on styles/blocks/staff-panel/staff-panel.css:
```diff
@@ -0,0 +1,29 @@
+    justify-content: space-between;
+}
+
+.staff-panel__grid-1r-1c {
```
Instead of presentational or cryptic names, always use ID and class names that reflect the purpose of the element in question, or that are otherwise generic.
Names that are specific and reflect the purpose of the element should be preferred as these are most understandable and the least likely to change.

That means that using `grid-1r-1c` is basically not allowed.

AnnGabr committed:
Removed grid from staff-panel block.

ahryhoryeu commented on styles/blocks/staff-panel/person.css:
```diff
@@ -0,0 +1,35 @@
+.person__photo {
+    float: left;
```
Since its container lacks `clearfix`, it doesn't scale to wrap its children size. Please make sure to add `clearfix` when using float or opt for inline-blocks, flex, etc. In this case it seems to me that there's no need for the image to float in the text.

ahryhoryeu commented on styles/blocks/attention/attention.css:
```diff
@@ -0,0 +1,6 @@
+.attention {
+    box-sizing: border-box;
```
This seems to be redundant.

ahryhoryeu commented on styles/blocks/attention/article.css:
```diff
@@ -0,0 +1,28 @@
+    max-width: 72%;
+    margin: auto;
+
+    text-align: center;
```
Having `margin: auto` allows you to avoid specifying `text-align` in this case.

ahryhoryeu commented on styles/blocks/company-life/life.css:
```diff
@@ -0,0 +1,26 @@
+    justify-content: space-between;
+}
+
+.life__grid-1r {
```
While using grid here is kind of justified, I want to emphasize that the classes should named based on semantics. So you need to come up with more meaningful names here.

ahryhoryeu commented on styles/blocks/company-life/life.css:
```diff
@@ -0,0 +1,26 @@
+}
+
+.life__grid-2r-1c {
+    grid-column: 1;
```
Specifying this is redundant since it follows the natural flow of grid elements.

ahryhoryeu commented on index.html:
```diff
@@ -5,296 +5,307 @@
+                <div class="life__grid-1r">
+                    <div class="life__header">
+                        <h2 class="h1-like life__title">Vestibulum Commodo Felis</h2>
+                        <p class="prh life__prh">Morbi interdum mollis sapien. Sed ac risus. Phasellus lacinia, magna</p>
```
From the semantics standpoint, it may even be a span.

AnnGabr replied:
Okey, in this case I had to set some display prop as block or inline-blocke to make line-height work, right?

AnnGabr committed:
Header fixed:
- changed nav-item--first to --active;
- removed fractions of pixels.

AnnGabr committed:
Removed flex from banner.

AnnGabr committed:
Removed paddings from service icons.

AnnGabr committed:
Removed negative margin.

AnnGabr committed:
Some classes renamed.

AnnGabr committed:
Refactored attention block.

AnnGabr committed:
Renamed grid inner classes in company-life block.

AnnGabr committed:
Moved header element to life component.

AnnGabr committed:
Changed p tag in life header to span.

AnnGabr committed:
Added block banner-article positioned by banner__article element.

AnnGabr committed:
Changed sizes of svg service icons.

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+            <ul class="footer__list">
+                <li class="footer-item">
+                    <h3 class="h3-like footer-item__title"><a class="link" href="#">Vitae lin</a></h3>
+                    <p class="footer-item__prh">Phasellus ultrices<br> Sed adipiscing lipsum<br>Nulla sed leoniton</p>
```
Actually this part of footer consists of groups (columns) of header and 3 links.

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+                            <div class="service-item__more"><a class="link" href="#">Read More<span class="service-item__more-icon icon-arrow-circle-right"/></a></div>
+                        </div>
+                    </li>
+                    <li class="service-item servise__list-item">
```
Fix a typo "servise" => "service."

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+                        </div>
+                        <div class="service-item__article">
+                            <h3 class="h3-like service-item__title">Cras ornare</h3>
+                            <p lang="en" class="prh service-item__prh">Simasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique</p>
```
JFYI. Attribute __lang__ is not required. There is no need to use it here.

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+                        <div class="service-item__article">
+                            <h3 class="h3-like service-item__title">Cras ornare</h3>
+                            <p lang="en" class="prh service-item__prh">Simasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique</p>
+                            <div class="service-item__more"><a class="link" href="#">Read More<span class="service-item__more-icon icon-arrow-circle-right"/></a></div>
```
For better readability try to avoid long lines. Foe example move tag __a__ to a new line.

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+                        </div>
+                    </li>
+                </ul>
+                <div class="tab about-us__tab">
```
I consider this block as something separate from "about us" section. It is just some sidebar information. There is one HTML elements that matches better for that purpose. :)
Also I think that it's worth to wrap 2 elements up with one parent. It will be better from semantic point of view.

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+                            <p class="says-item__who">- John Doe</p>
+                        </div>
+                        <div class="circle-icon says-item__icon">
+                            <img class="says-item__icon-img" src="./resources/images/comments.svg">
```
Always be consistent! Use some one format for src paths. Earlier you used just "resources/...".

drodov commented on index.html:
```diff
@@ -0,0 +1,299 @@
+                    </li>
+                </ul>
+            </section>
+            <section class="about-us">
```
There are several good candidates for HTML blocks _article_. Let's discuss offline what you think could be replaced with _article_.

AnnGabr committed:
Fixed "servise" => service.

AnnGabr committed:
Fixed long lines.

AnnGabr committed:
Added footer links.

AnnGabr committed:
Removed not required lang="en".

AnnGabr committed:
Fixed style of file pathes.

AnnGabr committed:
Changed div tags to article where needed.

AnnGabr committed:
Separated sidebar and about-us block.

AnnGabr committed:
Fixed page background.

AnnGabr committed:
Renamed "person__char" class.

AnnGabr committed:
Made page and banner block responsive.

AnnGabr committed:
Made service block responsive.

AnnGabr committed:
Made company-life block responsive.

AnnGabr committed:
Removed <br> from footer.

AnnGabr committed:
Removed <br> and wrap content with article tag where needed.

AnnGabr committed:
Added container class to site-name element.

AnnGabr committed:
Added before and after elements where needed.

AnnGabr committed:
Removed h-like classes.

AnnGabr committed:
Refactored:
- removed articles;
- removed normalize.css;
- changed names of selectors with "article";
- added title class as base for title modifiers.
